### PR TITLE
Win Support

### PR DESCRIPTION
**Pull request by steffengy**
Should be ready so far, not seeing anything that would prevent a merge.

## Post-Merge
- Push regenerated .h, .c and .go files (not included here to keep the diff clean)

### CODE REVIEW

File: libc/builtin.h [hunk 1]
```diff
@@ -18,7 +18,7 @@
 #define __builtin_prefetch(addr, ...) (void)(addr)
 #define __builtin_signbit(x) (sizeof(x) == sizeof(float) ? __signbitf(x) : sizeof (x) == sizeof(double) ? __signbit(x) : __signbitl(x))
 #define __builtin_types_compatible_p(type1, type2) __builtin_types_compatible__((type1){}, (type2){})
-#define __builtin_va_arg(ap, type) *(type*)(ap -= __roundup(sizeof(type), __SIZEOF_POINTER__))
+#define __builtin_va_arg(ap, type) *(type*)(ap -= (__UINTPTR_TYPE__)(__roundup(sizeof(type), __SIZEOF_POINTER__)))
```
Comment (cznic) [inline libc/builtin.h:21]: I guess this change is no more needed. But please feel free to keep it, it is correct anyway.

**Review by cznic — COMMENTED** [review]
One small suggestion.

File: libc/builtin.h (hunk 1 above)
Comment (steffengy) [reply libc/builtin.h:21]: Why is it not needed anymore?

Comment (cznic) [reply libc/builtin.h:21]: I'm not sure but I seem to recall this change predated fixing of the ptr1-ptr2 expression type from long to the proper platform-dependent type.

Comment (steffengy) [reply libc/builtin.h:21]: Getting a fault without it, keeping it.

Comment (cznic) [reply libc/builtin.h:21]: Proves I do not remember it correctly.

**Comment by steffengy** [comment]
Just curious & no rush if not: Have you already review this (andthe CC) PR or is that pending?

**Review by cznic — APPROVED** [review]
LGTM

**Comment by steffengy** [comment]
Added the regenerated .h and .c files for windows, let me know if I'm still good to merge

**Review by cznic — APPROVED** [review]
172749 lines added :open_mouth: 

I just take a look if it seems *structurally* sane, which seems to be the case, so assuming you've tested it works, it's LGTM.

**Comment by steffengy** [comment]
> 172749 lines added 😮
I just take a look if it seems structurally sane, which seems to be the case, so assuming you've tested it works, it's LGTM.

That's why I'm just adding it now, so you first could check the "real" changes :)
It's basically what I was using the whole time, just not commited until really necessary.

**Pull request merged**